---
title: "individual-crab-genes-over-time"
output: html_document
---
Rmd to get genes that change over time within individual crab RNA seq. 

Will start by making heatmaps of top genes (based on a cut-off threshold that `pheatmap` can handle in terms of number of rows) and then pull out the clusters of genes, and annotate. 

Using count data from transcriptome v 3.1.


### Load packages:
```{r}
library(dplyr)
library(tidyverse)
library(pheatmap)
library(data.table)
library(RColorBrewer)
```

## Read in datasets
Read in BLAST results from transcriptome v 3.1
```{r}
blast <- read.delim("../analyses/_blast-sep-v3.1.tab", header = FALSE)
head(blast)
```

Rename first and third column: 
```{r}
colnames(blast) <- c("Trinity_ID", "V2", "uniprot_acc_ID", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
head(blast)
```

Read in uniprot-SP-GO file
```{r}
uniprot_SP_GO <- read.delim("http://owl.fish.washington.edu/halfshell/bu-alanine-wd/17-07-20/uniprot-SP-GO.sorted", sep = '\t', header = FALSE)
head(uniprot_SP_GO)
```

Rename columns:
```{r}
colnames(uniprot_SP_GO) <- c("uniprot_acc_ID", "V2", "V3", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12")
head(uniprot_SP_GO)
```


### Load in data of counts from individual crab RNAseq (transcriptome v 3.1):   
```{r}
norm_counts <- read.delim("../data/salmon.isoform.TMM.EXPR.matrix")
head(norm_counts)
```

Set Trinity_ID as row names:
```{r}
rownames(norm_counts) <- norm_counts$X #set Trinity IDs as rownames
norm_counts <- norm_counts[,-1] #remove redundant column
head(norm_counts)
```

ROUND INTEGERS UP:
```{r}
norm_counts <- round(norm_counts,0)
head(norm_counts)
```

Set rownames as a column again:     
```{r}
norm_counts <- tibble::rownames_to_column(norm_counts, "Trinity_ID")
head(norm_counts)
```

Rename sample columns: 
```{r}
colnames(norm_counts) <- c("Trinity_ID", "CrabH_2", "CrabE_2", "CrabI_2", "CrabC_17", "CrabB_17", "CrabG_0", "CrabF_2", "CrabE_0", "CrabG_2", "CrabC_0", "CrabI_0", "CrabD_0", "CrabD_17", "CrabB_0", "CrabC_2", "CrabB_2", "CrabF_17", "CrabA_0", "CrabH_0", "CrabA_17", "CrabD_2", "CrabE_17", "CrabA_2", "CrabF_0")
head(norm_counts)
```

Set heatmap colors:
```{r}
heatmapBrBG <- RColorBrewer::brewer.pal(11, "BrBG")
```

# Crab A
### Subset samples from crab A (ambient infected)
```{r}
crabA <- select(norm_counts, "Trinity_ID", "CrabA_0", "CrabA_2", "CrabA_17")
head(crabA)
```

Wayyyy too many genes for a heatmap. Filter out rows that have 0 across all three timepoints:     
```{r}
crabA <- crabA[rowSums(crabA[, -1] >0) !=0, ]
head(crabA)
```
Now there's 16,427 genes 

Now lets remove low-count genes. Say, remove rows where counts across the three samples is less than 10. 
Sum all rows in new column:         
```{r}
crabA$sums <- rowSums( crabA[,2:4] )
head(crabA)
```

Remove columns with sums <= 30 --> didn't work. too many rows (3,504). I troubleshooted until coming to the number 55 - 2668 rows, and heatmap worked
```{r}
crabA_highcts <- filter(crabA, sums >= 55)
head(crabA_highcts)
```
Now there's 2,668 rows

### Make a heatmap of the list of 2668 genes:
Need to remove the "sums" column for heatmap:     
```{r}
crabA_heatmap <- select(crabA_highcts, "Trinity_ID", "CrabA_0", "CrabA_2", "CrabA_17")
head(crabA_heatmap)
```

Have to make trinity ID column into rownames:
Set Trinity_ID as row names:
```{r}
rownames(crabA_heatmap) <- crabA_heatmap$Trinity_ID #set Trinity IDs as rownames
crabA_heatmap <- crabA_heatmap[,-1] #remove redundant column
head(crabA_heatmap)
```

```{r}
pheatmap(crabA_heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```

Goal: pull out clusters of genes and describe and annotate. For example, cluster 1 --> in crab A goes zero, up, down over time. Then describe those genes (`join` with BLAST output from transcriptome v 3.1 and uniprot-sprot database info.)

### Get genes in those clusters that change over time:
Set heatmap as an object:
```{r}
Aheatmap <- pheatmap(crabA_heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```

Extract clusters from the heatmap. 
```{r}
CrabA.clust <- cbind(crabA_heatmap,
                   cluster = cutree(Aheatmap$tree_row,
                                     k = 4))
head(CrabA.clust)
```

pull out just cluster column:
```{r}
clusters <- select(CrabA.clust, "cluster")
head(clusters)
```

Rename the clusters in the rows to "cluster #" instead of just the number. 
```{r}
clusters <- data.frame(cluster = ifelse(clusters == 1, "cluster 1", ifelse(clusters == 2, "cluster 2", ifelse(clusters == 3, "cluster 3", "cluster 4"))))
head(clusters)
```

```{r}
pheatmap(crabA_heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_row = clusters, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```



### `join` sample count data for Crab A with transcriptome v 3.1 `blast` output and uniprot-sp-GO info, and keep the column of cluster labels. Write out to analyses. 

CrabA.clust --> heatmap counts with Trinity IDs as row names with cluster numbers. 

`join` blast output and uniprot-SP-GO:
```{r}
blastGO <- left_join(blast, uniprot_SP_GO, by = "uniprot_acc_ID")
head(blastGO)
```

Make Trinity ID rownames into a column called "Trinity_ID" in CrabA.clust:
Set Trinity_ID as row names:
Set rownames as a column again:     
```{r}
CrabA.clust <- tibble::rownames_to_column(CrabA.clust, "Trinity_ID")
head(CrabA.clust)
```

`join` the CrabA.clust and blastGO file based on "Trinity_ID" column:
```{r}
crabA.blast.GO <- left_join(CrabA.clust, blastGO, by = "Trinity_ID")
head(crabA.blast.GO)
```

write out as a table:
```{r}
#write.table(crabA.blast.GO, "../analyses/crabA.clust.BLAST-GO.tab", sep = '\t', row.names = FALSE, quote = FALSE)
```
Comment out 8/6/2020 because already wrote out file to analyses. 

# Crab B
## Subset samples for Crab B (ambient infected)
```{r}
crabB <- select(norm_counts, "Trinity_ID", "CrabB_0", "CrabB_2", "CrabB_17")
head(crabB)
```

Wayyyy too many genes for a heatmap. Filter out rows that have 0 across all three timepoints:     
```{r}
crabB <- crabB[rowSums(crabB[, -1] >0) !=0, ]
head(crabB)
```

Now lets remove low-count genes. Say, remove rows where counts across the three samples is less than 10. 
Sum all rows in new column:         
```{r}
crabB$sums <- rowSums( crabB[,2:4] )
head(crabB)
```

Remove columns with sums <= 30 --> didn't work. too many rows (3,504). I troubleshooted until coming to the number 55 - 2668 rows, and heatmap worked
```{r}
crabB_highcts <- filter(crabB, sums >= 28)
head(crabB_highcts)
```

### Make a heatmap of the list of 2668 genes:
Need to remove the "sums" column for heatmap:     
```{r}
crabB_heatmap <- select(crabB_highcts, "Trinity_ID", "CrabB_0", "CrabB_2", "CrabB_17")
head(crabB_heatmap)
```

Have to make trinity ID column into rownames:
Set Trinity_ID as row names:
```{r}
rownames(crabB_heatmap) <- crabB_heatmap$Trinity_ID #set Trinity IDs as rownames
crabB_heatmap <- crabB_heatmap[,-1] #remove redundant column
head(crabB_heatmap)
```

```{r}
pheatmap(crabB_heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```
6617 genes

# Crab C
## Subset samples from Crab C (ambient infected)

```{r}
crabC <- select(norm_counts, "Trinity_ID", "CrabC_0", "CrabC_2", "CrabC_17")
head(crabC)
```

Wayyyy too many genes for a heatmap. Filter out rows that have 0 across all three timepoints:     
```{r}
crabC <- crabC[rowSums(crabC[, -1] >0) !=0, ]
head(crabC)
```

Now lets remove low-count genes. Say, remove rows where counts across the three samples is less than 10. 
Sum all rows in new column:         
```{r}
crabC$sums <- rowSums( crabC[,2:4] )
head(crabC)
```

Remove columns with sums <= 30 --> didn't work. too many rows (3,504). I troubleshooted until coming to the number 55 - 2668 rows, and heatmap worked
```{r}
crabC_highcts <- filter(crabC, sums >= 49)
head(crabC_highcts)
```

### Make a heatmap of the list of 2668 genes:
Need to remove the "sums" column for heatmap:     
```{r}
crabC_heatmap <- select(crabC_highcts, "Trinity_ID", "CrabC_0", "CrabC_2", "CrabC_17")
head(crabC_heatmap)
```

Have to make trinity ID column into rownames:
Set Trinity_ID as row names:
```{r}
rownames(crabC_heatmap) <- crabC_heatmap$Trinity_ID #set Trinity IDs as rownames
crabC_heatmap <- crabC_heatmap[,-1] #remove redundant column
head(crabC_heatmap)
```

```{r}
pheatmap(crabC_heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```
2384 genes

# Crab D
## Subset samples for Crab D (decreased infected)

```{r}
crabD <- select(norm_counts, "Trinity_ID", "CrabD_0", "CrabD_2", "CrabD_17")
head(crabD)
```

Wayyyy too many genes for a heatmap. Filter out rows that have 0 across all three timepoints:     
```{r}
crabD <- crabD[rowSums(crabD[, -1] >0) !=0, ]
head(crabD)
```

Now lets remove low-count genes. Say, remove rows where counts across the three samples is less than 10. 
Sum all rows in new column:         
```{r}
crabD$sums <- rowSums( crabD[,2:4] )
head(crabD)
```

Remove columns with sums <= 30 --> didn't work. too many rows (3,504). I troubleshooted until coming to the number 55 - 2668 rows, and heatmap worked
```{r}
crabD_highcts <- filter(crabD, sums >= 0)
head(crabD_highcts)
```

### Make a heatmap of the list of 2668 genes:
Need to remove the "sums" column for heatmap:     
```{r}
crabD_heatmap <- select(crabD_highcts, "Trinity_ID", "CrabD_0", "CrabD_2", "CrabD_17")
head(crabD_heatmap)
```

Have to make trinity ID column into rownames:
Set Trinity_ID as row names:
```{r}
rownames(crabD_heatmap) <- crabD_heatmap$Trinity_ID #set Trinity IDs as rownames
crabD_heatmap <- crabD_heatmap[,-1] #remove redundant column
head(crabD_heatmap)
```

```{r}
pheatmap(crabD_heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```
can handle all genes. 7409 genes 

# Crab E
## subset samples for Crab E (decreased UNINFECTED) - only crab with individual RNAseq data that is UNINFECTED







