---
title: "Annotate individual crab gene count data with GO and BLAST output"
output: html_document
---
Rmd to annotate the list of individual crab RNAseq counts with uniprot-SP-GO.sorted and transcriptome v. 1.5 `blast` output: 

```{r}
library(tidyverse)
library(dplyr)
```

Read in 2020 Genewiz gene counts matrix (with log+1 counts as well): 
```{r}
indiv_counts <- read.delim("../analyses/individual_crab-isoform-log-matrix.txt", sep = '\t')
head(indiv_counts)
```

Rename columns: (only necessary if want to use the https://github.com/RobertsLab/paper-crab/blob/master/data/individual_crab-salmon.isoform.counts.matrix)
```{r}
#colnames(indiv_counts) <- c("Trinity_ID", "X294_quant", "X254_quant", "X280_quant", "X485_quant", "X481_quant", "X173_quant", "X222_quant", "X151_quant", "X272_quant", "X132_quant", "X127_quant", "X73_quant", "X427_quant", "X118_quant", "X334_quant", "X349_quant", "X425_quant", "X178_quant", "X72_quant", "X463_quant", "X221_quant", "X445_quant", "X359_quant", "X113_quant")
# head(indiv_counts)
```


Read in the uniprot-SP-GO.sorted file:
```{r}
uniprot_sp_GO <- read.delim("http://owl.fish.washington.edu/halfshell/bu-alanine-wd/17-07-20/uniprot-SP-GO.sorted", sep = '\t', header = FALSE)
head(uniprot_sp_GO)
```

Rename some columns: 
```{r}
colnames(uniprot_sp_GO) <- c("uniprot_acc_ID", "entry_name", "review_status", "protein_name", "gene_name", "organism", "length", "V8", "V9", "V10", "V11", "gene_ontology_IDs")
head(uniprot_sp_GO)
```

Upload crab transcriptome BLAST output (from transcriptome v. 1.5)
```{r}
blastout <- read.delim("../../project-crab/analyses/BLAST_to_GOslim/_blast-sep.tab", sep = '\t', header = FALSE)
head(blastout)
```


Rename one column so that this file can be `join`-ed with the uniprot-SP-Go:
```{r}
colnames(blastout) <- c("Trinity_ID", "V2", "uniprot_acc_ID", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
head(blastout)
```

`join` blastout and uniprot_sp_GO by "uniprot_acc_ID" column:
```{r}
blastout_uniprot <- left_join(blastout, uniprot_sp_GO, by = "uniprot_acc_ID")
head(blastout_uniprot)
```
Same rows as the `blast` output... good to go

`join` the annotated blast output file with individual crab count data by Trinity_ID:
```{r}
counts_annot <- left_join(indiv_counts, blastout_uniprot, by = "Trinity_ID")
counts_annot
```

Write out the table to paper-crab/analyses for use in making heatmaps: 
```{r}
write.table(counts_annot, "../analyses/individual-crab-counts-annot.tab", sep = '\t', row.names = FALSE, quote = FALSE)
```







