---
title: "Get genes in clusters"
output: html_document
---
Rmd to get the genes that are in the clusters of the heatmap comparing Crabs C and E across all time points with the infection DEG list. 

```{r}
library(pheatmap)
library(dplyr)
```

### Make heatmap in this Rmd to make things easier: 
Read in normalized counts: 
```{r}
norm_counts <- read.delim("../data/salmon.isoform.TMM.EXPR.matrix")
head(norm_counts)
```
40,035 rows of normalized count data for 24 samples

Rename columns: 
```{r}
colnames(norm_counts) <- c("Trinity_ID", "CrabH_2", "CrabE_2", "CrabI_2", "CrabC_17", "CrabB_17", "CrabG_0", "CrabF_2", "CrabE_0", "CrabG_2", "CrabC_0", "CrabI_0", "CrabD_0", "CrabD_17", "CrabB_0", "CrabC_2", "CrabB_2", "CrabF_17", "CrabA_0", "CrabH_0", "CrabA_17", "CrabD_2", "CrabE_17", "CrabA_2", "CrabF_0")
head(norm_counts)
```

Read in Temperature DEG list (772)
```{r}
infection_degs <- read.delim("../analyses/DEG-infection.txt", header = TRUE)
head(infection_degs)
```
`join` with norm_counts by "Trinity_ID"
```{r}
infection_degs_counts <- left_join(norm_counts, infection_degs, by = "Trinity_ID")
head(infection_degs_counts)
```

Just want the ones that match! (list of 772)
```{r}
infectdegs_counts_match <- filter(infection_degs_counts, baseMean != "NA")
head(infectdegs_counts_match)
```

**Crab C and E at day 2 and 17 with infection DEGs**
exclude day 0 because all crabs at ambient
_This one will be used for thesis defense_ 
Pull out just the sample count data:
```{r}
infectiondegs_CvE <- select(infectdegs_counts_match, Trinity_ID, CrabC_2, CrabC_17, CrabE_2, CrabE_17)
head(infectiondegs_CvE)
```

Set Trinity_ID as row names:
```{r}
rownames(infectiondegs_CvE) <- infectiondegs_CvE$Trinity_ID #set Trinity IDs as rownames
infectiondegs_CvE.heatmap <-infectiondegs_CvE[,-1] #remove redundant column
head(infectiondegs_CvE.heatmap)
```

I think I should try to remove rows that have 0 all the way across:
```{r}
noz_infectiondegs_CvE.heatmap <- infectiondegs_CvE.heatmap[rowSums(infectiondegs_CvE.heatmap[, -1] >0) !=0, ]
head(noz_infectiondegs_CvE.heatmap)
```
302 rows! 302 of the 772 infection degs are present in these samples. 

Create heatmap color scheme so that value of 0 is white:
```{r}
heatmapRedBlue <- RColorBrewer::brewer.pal(11, "RdBu")
```

```{r}
pheatmap(noz_infectiondegs_CvE.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapRedBlue, fontsize_col = 20, fontsize_row = 12)
```




## Get genes in the clusters
Set the heatmap as an object called "results": 

```{r}
results <- pheatmap(noz_infectiondegs_CvE.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapRedBlue, fontsize_col = 20, fontsize_row = 12)
```

# I think the following method works!!! 
Though it is kinda messy

https://stackoverflow.com/questions/27820158/pheatmap-in-r-how-to-get-clusters 

```{r}
results.clust <- cbind(noz_infectiondegs_CvE.heatmap, cluster = cutree(results$tree_row, k = 2))
head(results.clust)
```

    
The first gene "TRINITY_DN5315_c0_g1_i1" is only listed as being present in CrabC_17... and cluster designation is 1. So, Crab C is cluster1.      
I spot-checked a lot of other genes, and I think that cluster 1 is crab C, and cluster 2 is Crab E. 

## Get list of genes for each cluster
First: 
Set rownames as a column called Trinity ID
```{r}
res.clusts <- results.clust
names <- rownames(res.clusts)
rownames(res.clusts) <- NULL
final_clust_res <- cbind(names, res.clusts)
head(final_clust_res)
```
Rename column as Trinity ID
```{r}
colnames(final_clust_res) <- c("Trinity_ID", "CrabC_2", "CrabC_17", "CrabE_2", "CrabE_17", "cluster")
head(final_clust_res)
```

Pull out genes for each cluster: 
Cluster 1 --> Crab C at 2 and 17
```{r}
clust_crabC <- filter(final_clust_res, cluster == 1)
head(clust_crabC)
```
217 rows

Cluster 2 --> Crab E at 2 and 17
```{r}
clust_crabE <- filter(final_clust_res, cluster == 2)
head(clust_crabE)
```
85 rows

## Now let's figure out what those genes are doing. 

upload transcriptome v. 1.5 `blast` output:
```{r}
blastout <- read.delim("../../project-crab/analyses/BLAST_to_GOslim/_blast-sep.tab", sep = '\t', header = FALSE)
head(blastout)
```

Rename some columns so that this file can be `join`-ed with the uniprot-SP-Go:
```{r}
colnames(blastout) <- c("Trinity_ID", "V2", "uniprot_acc_ID", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
head(blastout)
```


Read in the uniprot-SP-GO.sorted file:
```{r}
uniprot_sp_GO <- read.delim("http://owl.fish.washington.edu/halfshell/bu-alanine-wd/17-07-20/uniprot-SP-GO.sorted", sep = '\t', header = FALSE)
head(uniprot_sp_GO)
```


Rename some columns: 
```{r}
colnames(uniprot_sp_GO) <- c("uniprot_acc_ID", "entry_name", "review_status", "protein_name", "gene_name", "organism", "length", "V8", "V9", "V10", "V11", "gene_ontology_IDs")
head(uniprot_sp_GO)
```


`join` blastout and uniprot_sp_GO by "uniprot_acc_ID" column:
```{r}
blastout_uniprot <- left_join(blastout, uniprot_sp_GO, by = "uniprot_acc_ID")
head(blastout_uniprot)
```

### `join` the `blast` output with uniprot info to the Crab C cluster genes:
```{r}
clust_crabC_annot <- left_join(clust_crabC, blastout_uniprot, by = "Trinity_ID")
head(clust_crabC_annot)
```

Just want to get an overall idea of what's going on with those genes, so remove those that weren't annotated: 

```{r}
clust_crabC_annot_match <- filter(clust_crabC_annot, entry_name != "NA")
head(clust_crabC_annot_match)
```
88 were annotated with `blast` and GO info. 

Write out this annotated cluster list. Will use uniprot accession IDs for enrichment in DAVID. 
```{r}
write.table(clust_crabC_annot_match, "../analyses/crabC-heatmap-cluster-genes.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```

### `join` the `blast` output with uniprot info to the Crab E cluster genes:
```{r}
clust_crabE_annot <- left_join(clust_crabE, blastout_uniprot, by = "Trinity_ID")
head(clust_crabE_annot)
```

Just want to get an overall idea of what's going on with those genes, so remove those that weren't annotated: 

```{r}
clust_crabE_annot_match <- filter(clust_crabE_annot, entry_name != "NA")
head(clust_crabE_annot_match)
```
41 rows. 

Write out this annotated cluster list. Will use uniprot accession IDs for enrichment in DAVID. 
```{r}
write.table(clust_crabE_annot_match, "../analyses/crabE-heatmap-cluster-genes.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```

