---
title: "Get genes in clusters"
output: html_document
---
Rmd to get the genes that are in the clusters of the heatmap comparing Crabs C and E across all time points with the infection DEG list. 

```{r}
#install.packages("plotrix")
library(plotrix)
```

```{r}
#install.packages("heatmap3")
library(heatmap3)
```


```{r}
library(pheatmap)
library(dplyr)
```

### Make heatmap in this Rmd to make things easier: 
Read in normalized counts: 
```{r}
norm_counts <- read.delim("../data/salmon.isoform.TMM.EXPR.matrix")
head(norm_counts)
```
40,035 rows of normalized count data for 24 samples

Rename columns: 
```{r}
colnames(norm_counts) <- c("Trinity_ID", "CrabH_2", "CrabE_2", "CrabI_2", "CrabC_17", "CrabB_17", "CrabG_0", "CrabF_2", "CrabE_0", "CrabG_2", "CrabC_0", "CrabI_0", "CrabD_0", "CrabD_17", "CrabB_0", "CrabC_2", "CrabB_2", "CrabF_17", "CrabA_0", "CrabH_0", "CrabA_17", "CrabD_2", "CrabE_17", "CrabA_2", "CrabF_0")
head(norm_counts)
```

Read in Temperature DEG list (772)
```{r}
infection_degs <- read.delim("../analyses/DEG-infection.txt", header = TRUE)
head(infection_degs)
```
`join` with norm_counts by "Trinity_ID"
```{r}
infection_degs_counts <- left_join(norm_counts, infection_degs, by = "Trinity_ID")
head(infection_degs_counts)
```

Just want the ones that match! (list of 772)
```{r}
infectdegs_counts_match <- filter(infection_degs_counts, baseMean != "NA")
head(infectdegs_counts_match)
```

**Crab C and E at day 2 and 17 with infection DEGs**
exclude day 0 because all crabs at ambient
_This one will be used for thesis defense_ 
Pull out just the sample count data:
```{r}
infectiondegs_CvE <- select(infectdegs_counts_match, Trinity_ID, CrabC_2, CrabC_17, CrabE_2, CrabE_17)
head(infectiondegs_CvE)
```

Set Trinity_ID as row names:
```{r}
rownames(infectiondegs_CvE) <- infectiondegs_CvE$Trinity_ID #set Trinity IDs as rownames
infectiondegs_CvE.heatmap <-infectiondegs_CvE[,-1] #remove redundant column
head(infectiondegs_CvE.heatmap)
```

I think I should try to remove rows that have 0 all the way across:
```{r}
noz_infectiondegs_CvE.heatmap <- infectiondegs_CvE.heatmap[rowSums(infectiondegs_CvE.heatmap[, -1] >0) !=0, ]
head(noz_infectiondegs_CvE.heatmap)
```
302 rows! 302 of the 772 infection degs are present in these samples. 

Rename Crab C and E to better names:
```{r}
colnames(noz_infectiondegs_CvE.heatmap) <- c("A_day2", "A_day17", "B_day2", "B_day17")
head(noz_infectiondegs_CvE.heatmap)
```

Create heatmap color scheme so that value of 0 is white:
```{r}
heatmapRedBlue <- RColorBrewer::brewer.pal(11, "RdBu")
```

```{r}
pheatmap(noz_infectiondegs_CvE.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapRedBlue, fontsize_col = 20, fontsize_row = 12)
```




## Get genes in the clusters
Set the heatmap as an object called "results": 

```{r}
results <- pheatmap(noz_infectiondegs_CvE.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapRedBlue, fontsize_col = 20, fontsize_row = 12)
```

```{r}
head(results)
```

```{r}
res_treerow <- results$tree_row
head(res_treerow)
```

```{r}
lbl <- cutree(res_treerow, 5)
head(lbl)
```

# I think the following method works!!! 
Though it is kinda messy

https://stackoverflow.com/questions/27820158/pheatmap-in-r-how-to-get-clusters 

```{r}
results.clust <- cbind(noz_infectiondegs_CvE.heatmap, cluster = cutree(results$tree_row, k = 2))
head(results.clust)
```

    
The first gene "TRINITY_DN5315_c0_g1_i1" is only listed as being present in CrabC_17... and cluster designation is 1. So, Crab C is cluster1.      
I spot-checked a lot of other genes, and I think that cluster 1 is crab C, and cluster 2 is Crab E. 

## Get list of genes for each cluster
First: 
Set rownames as a column called Trinity ID
```{r}
res.clusts <- results.clust
names <- rownames(res.clusts)
rownames(res.clusts) <- NULL
final_clust_res <- cbind(names, res.clusts)
head(final_clust_res)
```
Rename column as Trinity ID
```{r}
colnames(final_clust_res) <- c("Trinity_ID", "CrabC_2", "CrabC_17", "CrabE_2", "CrabE_17", "cluster")
head(final_clust_res)
```

Pull out genes for each cluster: 
Cluster 1 --> Crab C at 2 and 17
```{r}
clust_crabC <- filter(final_clust_res, cluster == 1)
head(clust_crabC)
```
217 rows

Cluster 2 --> Crab E at 2 and 17
```{r}
clust_crabE <- filter(final_clust_res, cluster == 2)
head(clust_crabE)
```
85 rows

## Now let's figure out what those genes are doing. 

upload transcriptome v. 1.5 `blast` output:
```{r}
blastout <- read.delim("../../project-crab/analyses/BLAST_to_GOslim/_blast-sep.tab", sep = '\t', header = FALSE)
head(blastout)
```

Rename some columns so that this file can be `join`-ed with the uniprot-SP-Go:
```{r}
colnames(blastout) <- c("Trinity_ID", "V2", "uniprot_acc_ID", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
head(blastout)
```


Read in the uniprot-SP-GO.sorted file:
```{r}
uniprot_sp_GO <- read.delim("http://owl.fish.washington.edu/halfshell/bu-alanine-wd/17-07-20/uniprot-SP-GO.sorted", sep = '\t', header = FALSE)
head(uniprot_sp_GO)
```


Rename some columns: 
```{r}
colnames(uniprot_sp_GO) <- c("uniprot_acc_ID", "entry_name", "review_status", "protein_name", "gene_name", "organism", "length", "V8", "V9", "V10", "V11", "gene_ontology_IDs")
head(uniprot_sp_GO)
```


`join` blastout and uniprot_sp_GO by "uniprot_acc_ID" column:
```{r}
blastout_uniprot <- left_join(blastout, uniprot_sp_GO, by = "uniprot_acc_ID")
head(blastout_uniprot)
```

### `join` the `blast` output with uniprot info to the Crab C cluster genes:
```{r}
clust_crabC_annot <- left_join(clust_crabC, blastout_uniprot, by = "Trinity_ID")
head(clust_crabC_annot)
```

Just want to get an overall idea of what's going on with those genes, so remove those that weren't annotated: 

```{r}
clust_crabC_annot_match <- filter(clust_crabC_annot, entry_name != "NA")
head(clust_crabC_annot_match)
```
88 were annotated with `blast` and GO info. 

Write out this annotated cluster list. Will use uniprot accession IDs for enrichment in DAVID. 
```{r}
write.table(clust_crabC_annot_match, "../analyses/crabC-heatmap-cluster-genes.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```

### `join` the `blast` output with uniprot info to the Crab E cluster genes:
```{r}
clust_crabE_annot <- left_join(clust_crabE, blastout_uniprot, by = "Trinity_ID")
head(clust_crabE_annot)
```

Just want to get an overall idea of what's going on with those genes, so remove those that weren't annotated: 

```{r}
clust_crabE_annot_match <- filter(clust_crabE_annot, entry_name != "NA")
head(clust_crabE_annot_match)
```
41 rows. 

Write out this annotated cluster list. Will use uniprot accession IDs for enrichment in DAVID. 
```{r}
write.table(clust_crabE_annot_match, "../analyses/crabE-heatmap-cluster-genes.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```


### Try Shelly's method: 

Lines 158-195: https://github.com/shellytrigg/paper-OysterSeed-TimeXTemp/blob/master/Analyses/PCA_ASCA_Functional_Analysis/PCA_ASCA_Functional_Analysis.Rmd 

Create a dendogram object: 
```{r}
hm <- as.dist(1-cor(as.matrix(t(noz_infectiondegs_CvE.heatmap), use = "pa")))
hm2 <- hclust(hm, method = 'average')
plot(hm2, cex=0.3)
abline(h=1, col = "red")
```



create a list of proteins in a cut dendrogram. 
```{r}
mycl <- cutree(hm2, h=0.01)
head(mycl)
```

Seems promising... 

Create a vector of colors that is length of number of clusters defined:
```{r}
clusterCols <- RColorBrewer::brewer.pal(4, "Pastel1")

```

create a vector of colors that mathces the list of genes where each gene is converted into a cluster color:
```{r}
myClusterSideBar <- clusterCols[mycl]
head(myClusterSideBar)
```


Then cluster color names can be extracted from the color ids using the color.id function. First create an object that is a vector of all color names:
```{r}
#get color name
clusterColor <- lapply(myClusterSideBar, color.id)
clusterColor <- lapply(clusterColor, `[[`,1)
clusterColor <- data.frame(unlist(clusterColor), stringsAsFactors = FALSE)
```


Create a datafram that contains cluster ID number, the cluster color, and the gene:
```{r}
foo <- cbind(data.frame(mycl, clusterColor))
#add gene column to data fram for merging
foo$Trinity_ID <- row.names(foo)
head(foo)
```

Merge the cluster info with the counts data by "Trinity_ID"
```{r}
#add the trinity id column back to the heatmap matrix:
noz_infectiondegs_CvE.heatmap_TrinityID <- data.frame(t(noz_infectiondegs_CvE.heatmap))
noz_infectiondegs_CvE.heatmap_TrinityID$Trinity_ID <- rownames(noz_infectiondegs_CvE.heatmap_TrinityID)
clade.genes <- merge(foo, noz_infectiondegs_CvE.heatmap_TrinityID, by = "Trinity_ID")
head(clade.genes)
```

Then plot the heatmap with the colored row side bars
```{r}
heatmap3(as.matrix(t(noz_infectiondegs_CvE.heatmap)), Colv = NA, cexRow = 0.5,labCol = c(19,21,21,23,23,25,25,27,27,29,29,31,31), RowSideColors=myClusterSideBar, ColAxisColors = 1,RowAxisColors=1,distfun = function(x) as.dist(1 - cor(t(x), use = "pa")), method = "average")

```

# Try to get 4 clusters using method earlier from stack overflow:

```{r}
results.clust4 <- cbind(noz_infectiondegs_CvE.heatmap, cluster = cutree(results$tree_row, k = 4))
head(results.clust4)
```

## Get list of genes for each cluster
First: 
Set rownames as a column called Trinity ID
```{r}
res.clusts4 <- results.clust4
names <- rownames(res.clusts4)
rownames(res.clusts4) <- NULL
final_clust_res4 <- cbind(names, res.clusts4)
head(final_clust_res4)
```

Rename column as Trinity ID
```{r}
colnames(final_clust_res4) <- c("Trinity_ID", "CrabC_2", "CrabC_17", "CrabE_2", "CrabE_17", "cluster")
head(final_clust_res4)
```

Look at final_clust_res4 in other tab. Match cluster number to sample ID based on the bigger normalized count for that trinity Id. 

Cluster 1 <-- Crab C _ 17

Cluster 2 <- Crab E _ 2

Cluster 3 --> Crab E_ 17

Cluster 4 --> Crab C _ 2 

Get those lists of Trinity IDs: 
Crab C _ 2:
```{r}
crabC2_clusgenes <- filter(final_clust_res4, cluster == 4)
head(crabC2_clusgenes)
```
41 rows

Crab C _ 17:
```{r}
crabC17_clusgenes <- filter(final_clust_res4, cluster == 1)
head(crabC17_clusgenes)
```
176 rows

Crab E _ 2:
```{r}
crabE2_clusgenes <- filter(final_clust_res4, cluster == 2)
head(crabE2_clusgenes)
```
31 rows

Crab E _ 17:
```{r}
crabE17_clusgenes <- filter(final_clust_res4, cluster == 3)
head(crabE17_clusgenes)
```
54 rows

## Get GO annotation for each cluster gene list and write out to analyses:
upload transcriptome v. 1.5 `blast` output:
```{r}
blastout <- read.delim("../../project-crab/analyses/BLAST_to_GOslim/_blast-sep.tab", sep = '\t', header = FALSE)
head(blastout)
```
Rename some columns so that this file can be `join`-ed with the uniprot-SP-Go:
```{r}
colnames(blastout) <- c("Trinity_ID", "V2", "uniprot_acc_ID", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
head(blastout)
```

Read in the uniprot-SP-GO.sorted file:
```{r}
uniprot_sp_GO <- read.delim("http://owl.fish.washington.edu/halfshell/bu-alanine-wd/17-07-20/uniprot-SP-GO.sorted", sep = '\t', header = FALSE)
head(uniprot_sp_GO)
```

Rename some columns: 
```{r}
colnames(uniprot_sp_GO) <- c("uniprot_acc_ID", "entry_name", "review_status", "protein_name", "gene_name", "organism", "length", "V8", "V9", "V10", "V11", "gene_ontology_IDs")
head(uniprot_sp_GO)
```

`join` blastout and uniprot_sp_GO by "uniprot_acc_ID" column:
```{r}
blastout_uniprot <- left_join(blastout, uniprot_sp_GO, by = "uniprot_acc_ID")
head(blastout_uniprot)
```

### Go for Crab C _2
```{r}
crabC2_GO <- left_join(crabC2_clusgenes, blastout_uniprot, by = "Trinity_ID")
head(crabC2_GO)
```

just save ones that matched:
```{r}
crabC2_GO_match <- filter(crabC2_GO, entry_name != "NA")
head(crabC2_GO_match)
```
22 rows

Write out table to analyses:
```{r}
write.table(crabC2_GO_match, "../analyses/crabC2_clustgenes_GO.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```

### Crab C 17 Go info:
```{r}
crabC17_GO <- left_join(crabC17_clusgenes, blastout_uniprot, by = "Trinity_ID")
head(crabC17_GO)
```
just save ones that matched:
```{r}
crabC17_GO_match <- filter(crabC17_GO, entry_name != "NA")
head(crabC17_GO_match)
```
66 rows 

Write out table to analyses:
```{r}
write.table(crabC17_GO_match, "../analyses/crabC17_clustgenes_GO.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```

### Crab E 2 Go:
```{r}
crabE2_GO <- left_join(crabE2_clusgenes, blastout_uniprot, by = "Trinity_ID")
head(crabE2_GO)
```

just save ones that matched:
```{r}
crabE2_GO_match <- filter(crabE2_GO, entry_name != "NA")
head(crabE2_GO_match)
```
15 rows

Write out table to analyses:
```{r}
write.table(crabE2_GO_match, "../analyses/crabE2_clustgenes_GO.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```

### Crab E 17 Go:
```{r}
crabE17_GO <- left_join(crabE17_clusgenes, blastout_uniprot, by = "Trinity_ID")
head(crabE17_GO)
```

just save ones that matched:
```{r}
crabE17_GO_match <- filter(crabE17_GO, entry_name != "NA")
head(crabE17_GO_match)
```
26 rows

Write out table to analyses:
```{r}
write.table(crabE17_GO_match, "../analyses/crabE17_clustgenes_GO.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```




