---
title: "Gene expression heatmaps - practice"
output: html_document
---
Rmd to create heatmaps of differentially expressed gene lists. Two DEG lists (with gene counts):   
1. Temperature DEG and gene count list:
https://raw.githubusercontent.com/RobertsLab/paper-crab/master/analyses/temp_DEGlist-counts.tab 
2. Infection DEG and gene count list: https://raw.githubusercontent.com/RobertsLab/project-crab/master/analyses/2019-infection-degs_genecounts.csv 

ALSO --> after those two heatmaps, I'm making a TON of heatmaps from the 2020 Genewiz individual RNA seq data. 

### Set up R Markdown document

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Install packages:
(uncomment "install.packages("pheatmap")" if you need to install it)
```{r}
#install.packages("pheatmap")
library(pheatmap)
```

```{r}
library(dplyr)
```

```{r}
#install.packages("RColorBrewer")
library(RColorBrewer)
```

# Temperature comparison (elevated vs decreased) heatmap

Create color scheme and save as object:
(11 --> number of colors in color palette)
```{r}
heatmapGreens <- RColorBrewer::brewer.pal(9, "Greens")
```

### Read in temperature DEG list with gene counts table (423 rows)
```{r}
temp_counts <- read.delim("../analyses/temp_DEGlist-counts-log.txt", sep = '\t')
head(temp_counts)
```

Subset just the Trinity_ID column and the gene count sample columns (log +1 transformed):
```{r}
matrix_temp <- select(temp_counts, Trinity_ID, X380822_cold_log, X380823_cold_log, X380824_warm_log, X380825_warm_log)
head(matrix_temp)
```

Set Trinity_ID as row names:
```{r}
rownames(matrix_temp) <- matrix_temp$Trinity_ID #set Trinity_ID as rownames
matrix_temp.heatmap <- matrix_temp[,-1] #remove redundant column
head(matrix_temp.heatmap)
```

Subset genes of interest (for practice, just picking random ones)
```{r}
#include_list <- c("TRINITY_DN169_c0_g1_i11", "TRINITY_DN983_c0_g1_i3", "TRINITY_DN25158_c34_g1_i1", "TRINITY_DN5806_c0_g1_i1")
#matrix_temp.heatmap.sub <- matrix_temp.heatmap[include_list, ]
#matrix_temp.heatmap.sub
```

Create heatmap for temperature DEG list:
```{r}
pheatmap(matrix_temp.heatmap, cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = FALSE, treeheight_col = 80, legend = TRUE, color = heatmapGreens, fontsize_col = 20, fontsize_row = 12)
```


# Infection comparison heatmap 

Heatmap color palette: 
```{r}
heatmapOranges <- RColorBrewer::brewer.pal(9, "Oranges")
```

### Read in infection status DEG list with gene counts table (772 rows)
```{r}
infect_counts <- read.delim("../../project-crab/analyses/2019-infection-degs_genecounts.csv", sep = ',')
head(infect_counts)
```

Subset just the Trinity_ID column and the gene count sample columns:
```{r}
matrix_infect <- select(infect_counts, Trinity_ID, infected_12, uninfected_12, infected_26, uninfected_26)
head(matrix_infect)
```

Set Trinity_ID as row names:
```{r}
rownames(matrix_infect) <- matrix_infect$Trinity_ID #set Trinity_ID as rownames
matrix_infect.heatmap <- matrix_infect[,-1] #remove redundant column
head(matrix_infect.heatmap)
```

Subset genes of interest (for practice, just picking random ones)
```{r}
#include_list <- c("TRINITY_DN1003_c0_g1_i4", "TRINITY_DN7071_c0_g1_i1", "TRINITY_DN1796_c0_g1_i1", "TRINITY_DN1796_c0_g1_i1")
#matrix_infect.heatmap.sub <- matrix_infect.heatmap[include_list, ]
#matrix_infect.heatmap.sub
```

Create heatmap for temperature subset DEG list:
```{r}
pheatmap(matrix_infect.heatmap, cluster_row = FALSE, cluster_cols = TRUE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapOranges, fontsize_col = 20, fontsize_row = 12)
```

------------------------------------------------------------------------
# HEAT MAPS FOR INDIVIDUAL CRAB RNASEQ DATA
Just making a ton to see if anything is interesting 

upload individual RNAseq gene count data (2020 Genewiz samples): with log+1 transformation
```{r}
master_individual_counts <- read.delim("../analyses/individual-crab-isoform-counts-log.txt", sep = '\t')
head(master_individual_counts)
```

Pull out column Trinity IDs and all individual samples that have _log in column name: 
```{r}
matrix_individ_log <- select(master_individual_counts, X, X294_quant_log, X254_quant_log, X280_quant_log, X485_quant_log, X481_quant_log, X173_quant_log, X222_quant_log, X151_quant_log, X272_quant_log, X132_quant_log, X127_quant_log, X73_quant_log, X427_quant_log, X118_quant_log, X334_quant_log, X349_quant_log, X425_quant_log, X178_quant_log, X72_quant_log, X463_quant_log, X221_quant_log, X445_quant_log, X359_quant_log, X113_quant_log)
head(matrix_individ_log)
```

Rename columns based on crab ID along with sample day (example: A_0, A_2, A_17)
Tab 3 in this google doc give GeneWiz IDs for the individual crabs (A-I)
https://docs.google.com/spreadsheets/d/1d17yg5F5gKKC66O8QkTIlPxljJeuX7ZsG46pkBr1lNQ/edit#gid=848044428 

```{r}
colnames(matrix_individ_log) <- c("Trinity_ID", "CrabH_2", "CrabE_2", "CrabI_2", "CrabC_17", "CrabB_17", "CrabG_0", "CrabF_2", "CrabE_0", "CrabG_2", "CrabC_0", "CrabI_0", "CrabD_0", "CrabD_17", "CrabB_0", "CrabC_2", "CrabB_2", "CrabF_17", "CrabA_0", "CrabH_0", "CrabA_17", "CrabD_2", "CrabE_17", "CrabA_2", "CrabF_0")
head(matrix_individ_log)
```

# Heatmap for crabs A, B, and C over time
CrabA 0, 2, 17 (infected ambient)
CrabB 0, 2, 17 (infected ambient)
CrabC 0, 2, 17 (infected ambient)

Subset the Trinity IDs and the counts for those samples:
```{r}
ABC <- select(matrix_individ_log, Trinity_ID, CrabA_0, CrabA_2, CrabA_17, CrabB_0, CrabB_2, CrabB_17, CrabC_0, CrabC_2, CrabC_17)
head(ABC)
```

Set Trinity_ID as row names:
```{r}
rownames(ABC) <- ABC$Trinity_ID #set Trinity IDs as rownames
ABC.heatmap <- ABC[,-1] #remove redundant column
head(ABC.heatmap)
```

Create heatmap for temperature subset DEG list:
24,727 rows ---> TOO MANY FOR ONE HEATMAP

Open the annotated version of the individual crab isoform matrix to find genes of interest (compared to list of stress response genes)
```{r}
annotated <- read.delim("../analyses/individual-crab-counts-annot.tab", sep = '\t')
head(annotated)
```

Subset some genes (cross-references with stress response gene list): 
TRINITY_DN3510_c0_g1_i1 	DnaJ homolog subfamily B member 4 (protein folding [GO:0006457]; response to heat [GO:0009408];)

TRINITY_DN745_c3_g1_i2 proPO-A3 (defense response to fungus, response to bacterium, melanization defense response, etc)

TRINITY_DN1520_c0_g1_i1 HSP 90-beta (response to stress and lots of other stuff)

TRINITY_DN3772_c0_g1_i1 Heat shock protein 75 kDa, mitochondrial (HSP 75) (TNFR-associated protein 1) response to stress [GO:0006950]; extracellular exosome [GO:0070062]; negative regulation of cellular respiration [GO:1901856]; negative regulation of intrinsic apoptotic signaling pathway in response to hydrogen peroxide [GO:1903751]; response to stress [GO:0006950]; 
```{r}
include_list <- c("TRINITY_DN3510_c0_g1_i1", "TRINITY_DN745_c3_g2_i2", "TRINITY_DN1520_c0_g1_i1", "TRINITY_DN3772_c0_g1_i1")
ABC.heatmap.sub <- ABC.heatmap[include_list, ]
ABC.heatmap.sub
```

```{r}
pheatmap(ABC.heatmap.sub, cluster_row = FALSE, cluster_cols = TRUE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = TRUE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapOranges, fontsize_col = 20, fontsize_row = 12)
``` 


# HEAT MAPS FOR COMPARING CRAB D AND E
Comparing D and E over time
Crab D --> uninfected, decreased temp
Crab E --> infected, decreased temp


```{r}
DE <- select(matrix_individ_log, Trinity_ID, CrabD_0, CrabD_2, CrabD_17, CrabE_0, CrabE_2, CrabE_17, CrabE_0, CrabE_2, CrabE_17)
head(DE)
```

Set Trinity_ID as row names:
```{r}
rownames(DE) <- DE$Trinity_ID #set Trinity IDs as rownames
DE.heatmap <- DE[,-1] #remove redundant column
head(DE.heatmap)
```



```{r}
include_list <- c("TRINITY_DN3510_c0_g1_i1", "TRINITY_DN745_c3_g2_i2", "TRINITY_DN1520_c0_g1_i1", "TRINITY_DN3772_c0_g1_i1")
DE.heatmap.sub <- DE.heatmap[include_list, ]
DE.heatmap.sub
```

```{r}
pheatmap(DE.heatmap.sub, cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = TRUE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapGreens, fontsize_col = 20, fontsize_row = 12)
``` 

# Heatmap for crabs G, H, and I across time
Crab G - infected elevated
Crab H - infected elevated
Crab I - infected elevated 

```{r}
GHI <- select(matrix_individ_log, Trinity_ID, CrabG_0, CrabG_2, CrabH_0, CrabH_2, CrabI_0, CrabI_2)
head(GHI)
```

Set Trinity_ID as row names:
```{r}
rownames(GHI) <- GHI$Trinity_ID #set Trinity IDs as rownames
GHI.heatmap <- GHI[,-1] #remove redundant column
head(GHI.heatmap)
```

```{r}
include_list <- c("TRINITY_DN3510_c0_g1_i1", "TRINITY_DN745_c3_g2_i2", "TRINITY_DN1520_c0_g1_i1", "TRINITY_DN3772_c0_g1_i1")
GHI.heatmap.sub <- GHI.heatmap[include_list, ]
GHI.heatmap.sub
```

```{r}
pheatmap(GHI.heatmap.sub, cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = TRUE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapOranges, fontsize_col = 20, fontsize_row = 12)
```

# Heatmap for crabs A, E, and G across 2 time points 
Crab A - infected ambient
Crab E - infected decreased
Crab G - infected elevated

```{r}
AEG <- select(matrix_individ_log, Trinity_ID, CrabA_0, CrabA_2, CrabE_0, CrabE_2, CrabG_0, CrabG_2)
head(AEG)
```

Set Trinity_ID as row names:
```{r}
rownames(AEG) <- AEG$Trinity_ID #set Trinity IDs as rownames
AEG.heatmap <- AEG[,-1] #remove redundant column
head(AEG.heatmap)
```

Create heatmap colors:
```{r}
heatmapBlues <- RColorBrewer::brewer.pal(9, "Blues")
```


```{r}
include_list <- c("TRINITY_DN3510_c0_g1_i1", "TRINITY_DN745_c3_g2_i2", "TRINITY_DN1520_c0_g1_i1", "TRINITY_DN3772_c0_g1_i1")
AEG.heatmap.sub <- AEG.heatmap[include_list, ]
AEG.heatmap.sub
```

```{r}
pheatmap(AEG.heatmap.sub, cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = TRUE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBlues, fontsize_col = 20, fontsize_row = 12)
```

# Heat map for temeprature DEG list - with ABC and GHI 

`join` temperature DEG list with individual crab count matrix
```{r}
temp_individual <- left_join(matrix_individ_log, temp_counts, by = "Trinity_ID")
head(temp_individual)
```
Just want the ones that matched..

```{r}
tempdegs_individual <- filter(temp_individual, baseMean != "NA")
head(tempdegs_individual)
```

HEAT MAP FOR CRABS ABC AND GHI WITH TRINITY LIST FROM TEMPERATURE DEGS

```{r}
ABCGHI <- select(tempdegs_individual, Trinity_ID, CrabA_0, CrabA_2, CrabB_0, CrabB_2, CrabC_0, CrabC_2, CrabG_0, CrabG_2, CrabH_0, CrabH_2, CrabI_0, CrabI_2)
head(ABCGHI)
```

Create heatmap colors:
```{r}
heatmapReds <- RColorBrewer::brewer.pal(9, "Reds")
```

Set Trinity_ID as row names:
```{r}
rownames(ABCGHI) <- ABCGHI$Trinity_ID #set Trinity IDs as rownames
ABCGHI.heatmap <- ABCGHI[,-1] #remove redundant column
head(ABCGHI.heatmap)
```

```{r}
pheatmap(ABCGHI.heatmap, cluster_row = FALSE, cluster_cols = TRUE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapReds, fontsize_col = 20, fontsize_row = 12)
```
DEG temperature list that matched with the individual crab gene counts

# Heat maps with infection comparison DEG list and individual crabs

`join` infect_counts with matrix_individ_log
```{r}
infdegs_individual <- left_join(matrix_individ_log, infect_counts, by = "Trinity_ID")
head(infdegs_individual)
```

Just want the ones that matched..

```{r}
infected_individual_degs <- filter(infdegs_individual, baseMean != "NA")
head(infected_individual_degs)
```

```{r}
ABCGHI_infected <- select(infected_individual_degs, Trinity_ID, CrabA_0, CrabA_2, CrabB_0, CrabB_2, CrabC_0, CrabC_2, CrabG_0, CrabG_2, CrabH_0, CrabH_2, CrabI_0, CrabI_2)
head(ABCGHI_infected)
```

Set Trinity_ID as row names:
```{r}
rownames(ABCGHI_infected) <- ABCGHI_infected$Trinity_ID #set Trinity IDs as rownames
ABCGHIinfected.heatmap <- ABCGHI_infected[,-1] #remove redundant column
head(ABCGHIinfected.heatmap)
```

```{r}
pheatmap(ABCGHIinfected.heatmap, cluster_row = FALSE, cluster_cols = TRUE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapReds, fontsize_col = 20, fontsize_row = 12)
```

# Heat map of infected DEGs with crabs D and E across time points 
D --> uinfected decreased
E --> infected decreased

```{r}
DE_infected <- select(infected_individual_degs, Trinity_ID, CrabD_0, CrabD_2, CrabD_17, CrabE_0, CrabE_2, CrabE_17)
head(DE_infected)
```

Set Trinity_ID as row names:
```{r}
rownames(DE_infected) <- DE_infected$Trinity_ID #set Trinity IDs as rownames
DEinfected.heatmap <- DE_infected[,-1] #remove redundant column
head(DEinfected.heatmap)
```


```{r}
pheatmap(DEinfected.heatmap, cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapReds, fontsize_col = 20, fontsize_row = 12)
```
Crab D --> uninfected decreased
Crab E --> infected decreased


# Heat map of all individual crabs gene counts

```{r}
ALLcrabs <- select(matrix_individ_log, Trinity_ID, CrabA_0, CrabA_2, CrabA_17, CrabB_0, CrabB_2, CrabB_17, CrabC_0, CrabC_2, CrabC_17, CrabD_0, CrabD_2, CrabD_17, CrabE_0, CrabE_2, CrabE_17, CrabF_0, CrabF_2, CrabF_17, CrabG_0, CrabG_2, CrabH_0, CrabH_2, CrabI_0, CrabI_2)
head(ALLcrabs)
```

Read in the stress response gene list from transcriptome v. 1.5:
```{r}
stress <- read.delim("../../project-crab/analyses/crab_stress-response-genes.tab", sep = '\t')
head(stress)
```

`join` the stress response gene list with the Trinity IDs from all individual crabs: 
```{r}
stress_ALLcrabs <- left_join(ALLcrabs, stress, by = "Trinity_ID")
head(stress_ALLcrabs)
```

Subset just the ones that matched: 
```{r}
matched_stress_ALLcrabs <- filter(stress_ALLcrabs, GO_ID != "NA")
head(matched_stress_ALLcrabs)
```

Subset just the rows that Trinity IDs and the individual crabs: 
```{r}
stressALLcrabs <- select(matched_stress_ALLcrabs, Trinity_ID, CrabA_0, CrabA_2, CrabA_17, CrabB_0, CrabB_2, CrabB_17, CrabC_0, CrabC_2, CrabC_17, CrabD_0, CrabD_2, CrabD_17, CrabE_0, CrabE_2, CrabE_17, CrabF_0, CrabF_2, CrabF_17, CrabG_0, CrabG_2, CrabH_0, CrabH_2, CrabI_0, CrabI_2)
head(stressALLcrabs)
```

```{r}
stressALLcrabs_distinct <- distinct(stressALLcrabs)
head(stressALLcrabs_distinct)
```


Set Trinity_ID as row names:
```{r}
rownames(stressALLcrabs_distinct) <- stressALLcrabs_distinct$Trinity_ID #set Trinity IDs as rownames
stressALLcrabs_distinct.heatmap <- stressALLcrabs_distinct[,-1] #remove redundant column
head(stressALLcrabs_distinct.heatmap)
```
3033 rows


MAKING A HEAT MAP WITH THAT MANY GENES resulted in R and my laptop shutting down 

```{r}
pheatmap(stressALLcrabs_distinct.heatmap, cluster_row = FALSE, cluster_cols = TRUE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapGreens, fontsize_col = 20, fontsize_row = 12)
```

```{r}
pheatmap(stressALLcrabs_distinct.heatmap, cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapGreens, fontsize_col = 20, fontsize_row = 12)
```










