---
title: "DESeq-crabsABC-crabsGHI"
output: html_document
---
Rmd to use `DESeq2` to get differentially expressed gene list from comparing Crabs A, B, and C (infected, ambient - day 2) to Crabs G, H, and I (infected, elevated - day 2). Crabs A, B, and C will be combined into 1 group (infected, ambient, day 2) and crabs G, H, and I will be combined in 1 group (infected, elevated, day 2). 

```{r}
library(DESeq2)
```

```{r}
library(dplyr)
library(tidyverse)
```


Read in the isoform count matrix from the 2020 Genewiz individual crab RNAseq data (un-normalized!!):
```{r}
indiv_counts <- read.delim("../data/individual_crab-salmon.isoform.counts.matrix")
head(indiv_counts)
```

We only want day 2 samples from crabs A, B, C and G, H, I
Crab A day 2 - 359
Crab B day 2 - 349
Crab C day 2 - 334
Crab G day 2 - 272
Crab H day 2 - 294
Crab I day 2 - 280 

ALSO! Select the first column "X" with Trinity IDs

Subset those columns:
```{r}
abcghi <- select(indiv_counts, X, X359_quant, X349_quant, X334_quant, X272_quant, X294_quant, X280_quant)
head(abcghi)
```

Rename columns: 
```{r}
colnames(abcghi) <- c("X", "359_ambient", "349_ambient", "334_ambient", "272_elevated", "294_elevated", "280_elevated")
head(abcghi)
```

Make Column X (Trinity IDs) row names:
```{r}
rownames(abcghi) <- abcghi$X
abcghi <- abcghi[,-1]
head(abcghi)
```

Round counts up to integers!!!
```{r}
abcghi <- round(abcghi,0)
head(abcghi)
```

Rest is done mirroring what SR did for temperature comparison (02-DeSeq-Temperature.Rmd).        
Making a matrix for the comparison: 
```{r}
deseq2.colData <- data.frame(condition=factor(c(rep("Ambient", 3), rep("Elevated", 3))), type=factor(rep("paired-end", 6)))
rownames(deseq2.colData) <- colnames(data)
deseq2.dds <- DESeqDataSetFromMatrix(countData = abcghi,
                                     colData = deseq2.colData, 
                                     design = ~ condition)
```


```{r}
deseq2.dds <- DESeq(deseq2.dds)
deseq2.res <- results(deseq2.dds)
deseq2.res <- deseq2.res[order(rownames(deseq2.res)), ]
```

```{r}
head(deseq2.res)
```

```{r}
write.table(deseq2.res, "../analyses/deseq2.res-ABCGHI.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ])
```

```{r}
abcghi_fig <- deseq2.res
# The main plot
plot(abcghi_fig$baseMean, abcghi_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="ABC GHI  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
abcghi.sig <- deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ]
points(abcghi.sig$baseMean, abcghi.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

```{r}
write.table(abcghi.sig, "../analyses/DEG-ABCGHI.tab", row.names = T, quote = FALSE, sep = '\t')
```

```{bash}
head ../analyses/DEG-ABCGHI.tab
```
```{bash}
wc -l ../analyses/DEG-ABCGHI.tab
```

# Heatmap - try getting vst for counts
```{r}
library(pheatmap)
```

```{r}
#transform data 
vsd <- varianceStabilizingTransformation(deseq2.dds, blind = FALSE)#variance stabilizing transoformation
rld <- rlog(deseq2.dds, blind = FALSE) #controls for rows with few counts
ntd <- normTransform(deseq2.dds) #transforms data for plotting
top20 <- order(rowMeans(counts(deseq2.dds, normalized = TRUE)),
               decreasing = TRUE)[1:20] #average counts for each gene and select top 20 most abundant genes
pheatmap(assay(ntd)[top20,])
```

Try from DESeq2 manual: ntd heatmap
```{r}
select <- order(rowMeans(counts(deseq2.dds, normalized = TRUE)),
                decreasing = TRUE)[1:20]
df <- as.data.frame(colData(deseq2.dds)[,c("condition", "type")])
pheatmap(assay(ntd)[select,], cluster_rows = FALSE, show_rownames = FALSE, cluter_cols= FALSE, annotation_col = df)
```

vsd heatmap:
```{r}
pheatmap(assay(vsd)[select,], cluster_rows = FALSE, show_rownames = FALSE, cluster_cols = FALSE, annotation_col = df)
```
rld heatmap:
```{r}
pheatmap(assay(rld)[select,], cluster_rows = FALSE, show_rownames = FALSE, cluster_cols = FALSE, annotation_col = df)
```
Heatmap of sample to sample distances:
```{r}
sampleDists <- dist(t(assay(vsd)))
```

```{r}
library("RColorBrewer")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$condition, vsd$type, sep="-")
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
```

PCA 
```{r}
plotPCA(vsd, intgroup=c("condition","type"))
```

```{r}
pcaData <- plotPCA(vsd, intgroup=c("condition", "type"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=condition, shape=type)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()
``` 

Try to save the variance stabilized transformed data for use in other heatmaps: 
```{r}
ABCGHI_vst_degs <- getVarianceStabilizedData(deseq2.dds)
head(ABCGHI_vst_degs)
```

