---
title: "time-series"
output: html_document
---
Rmd for performing time-series analyses with the individual crab RNAseq data sets. 
https://www.bioconductor.org/help/course-materials/2015/LearnBioconductorFeb2015/B02.1.1_RNASeqLab.html#time 

```{r}
library(DESeq2)
library(tidyverse)
library(pheatmap)
```

## Read in and prepare dataset: 
FROM TRANSCRIPTOME V. 1.5
Read in the isoform count matrix from the 2020 Genewiz individual crab RNAseq data (un-normalized!!):
```{r}
indiv_counts <- read.delim("../data/individual_crab-salmon.isoform.counts.matrix")
head(indiv_counts)
```

Rename columns:
Crab Letter (A-I based on table in paper) and sample day. Table in paper will tell infection status of the crabs. 
```{r}
colnames(indiv_counts) <- c("X", "CrabH_2", "CrabE_2", "CrabI_2", "CrabC_17", "CrabB_17", "CrabG_0", "CrabF_2", "CrabE_0", "CrabG_2", "CrabC_0", "CrabI_0", "CrabD_0", "CrabD_17", "CrabB_0", "CrabC_2", "CrabB_2", "CrabF_17", "CrabA_0", "CrabH_0", "CrabA_17", "CrabD_2", "CrabE_17", "CrabA_2", "CrabF_0")
head(indiv_counts)
```

Make Column X (Trinity IDs) row names:
```{r}
rownames(indiv_counts) <- indiv_counts$X
indiv_counts <- indiv_counts[,-1]
head(indiv_counts)
```


Round counts up to integers!!!
```{r}
indiv_counts <- round(indiv_counts,0)
head(indiv_counts)
```

## `DESeq2` analyses on some samples:
I'm not going to do it on all samples at once. 

I'll start with ABC vs GHI time points 0 and 2. Kinda weird because at time 0 all samples are at ambient... but that's all I can do. 
Subset those columns:
```{r}
abcghi <- select(indiv_counts, CrabA_0, CrabB_0, CrabC_0, CrabA_2, CrabB_2, CrabC_2, CrabG_0, CrabH_0, CrabI_0, CrabG_2, CrabH_2, CrabI_2)
head(abcghi)
```


```{r}
abcghi.colData <- data.frame(temperature=factor(c(rep("ambient", 6), rep("elevated", 6))), 
                             day=factor(c(rep("day0", 3), rep("day2", 3), rep("day0", 3), rep("day2", 3))))
rownames(abcghi.colData) <- colnames(abcghi)
abcghi.dds <- DESeqDataSetFromMatrix(countData = abcghi,
                                     colData = abcghi.colData, 
                                     design = ~ temperature + day + temperature:day)
```

Check levels of temperature and day:
```{r}
levels(abcghi.dds$temperature)
```

```{r}
levels(abcghi.dds$day)
```

```{r}
design(abcghi.dds) <- formula(~ temperature + day)
abcghi.dds <- DESeq(abcghi.dds)
abcghi.res <- results(abcghi.dds)
abcghi.res$symbol <- mcols(abcghi.dds)$symbol
head(abcghi.res[order(abcghi.res$pvalue),],4)
```


```{r}
abcghi.res <- abcghi.res
# The main plot
plot(abcghi.res$baseMean, abcghi.res$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
abcghi.res.sig <- abcghi.res[!is.na(abcghi.res$padj) & abcghi.res$padj <= 0.05, ]
points(abcghi.res.sig$baseMean, abcghi.res.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

```{r}
#write.table(abcghi.res.sig, "../analyses/DEGlist-abcghi-temp_day.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```


```{r}
data <- plotCounts(abcghi.dds, which.min(abcghi.res$pvalue),
                   intgroup = c("day","temperature"), returnData = TRUE)
ggplot(data, aes(x=day, y=count, color = temperature, group=temperature)) + 
  geom_point() + stat_smooth(se=FALSE,method="loess") + scale_y_log10()
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(abcghi.res[!is.na(abcghi.res$padj) & abcghi.res$padj <= 0.05, ])
```
137 degs comparing time while taking temp into account 

Write out list of 137 DEGs:
```{r}
write.table(abcghi.res.sig, "../analyses/DEGlist-abcghi-daytemp.tab", sep = '\t', quote = FALSE, row.names = TRUE, col.names = TRUE)
```


## `join` with count data to make a heatmap:
read in .txt version
```{r}
daytemp_degs <- read.delim("../analyses/DEGlist-abcghi-daytemp.txt", sep = '\t')
head(daytemp_degs)
```

Make row names of abcghi Trinity IDs into a column called "Trinity_ID":
```{r}
#make Trinity IDs into a column in tmdf
df <- abcghi
names <- rownames(abcghi)
rownames(df) <- NULL
data <- cbind(names, df)

#rename column 1 in tmdf
colnames(data) <- c("Trinity_ID", "CrabA_0", "CrabB_0", "CrabC_0", "CrabA_2", "CrabB_2", "CrabC_2", "CrabG_0", "CrabH_0", "CrabI_0", "CrabG_2", "CrabH_2", "CrabI_2")
head(data)
```

`join` based on Trinity ID
```{r}
daytemp_degs_counts <- left_join(data, daytemp_degs, by = "Trinity_ID")
head(daytemp_degs_counts)
```

Only want the ones that match!
```{r}
daytemp_degs_counts_match <- filter(daytemp_degs_counts, baseMean != "NA")
head(daytemp_degs_counts_match)
```

Set Trinity_ID as row names:
```{r}
rownames(daytemp_degs_counts_match) <- daytemp_degs_counts_match$Trinity_ID #set Trinity IDs as rownames
daytemp_degs_counts_match.heatmap <- daytemp_degs_counts_match[,-1] #remove redundant column
head(daytemp_degs_counts_match.heatmap)
```


Pull out just the sample count data:
```{r}
daytemp_degs_counts_match.heatmap <- select(daytemp_degs_counts_match.heatmap, CrabA_0, CrabB_0, CrabC_0, CrabA_2, CrabB_2, CrabC_2, CrabG_0, CrabH_0, CrabI_0, CrabG_2, CrabH_2, CrabI_2)
head(daytemp_degs_counts_match.heatmap)
```

Going to remove all rows that have "0" for all samples: 
```{r}
noz_daytemp_degs_counts_match.heatmap <- daytemp_degs_counts_match.heatmap[rowSums(daytemp_degs_counts_match.heatmap[, -1] >0) !=0, ]
head(noz_daytemp_degs_counts_match.heatmap)
```

```{r}
heatmapPurpleOrange <- RColorBrewer::brewer.pal(11, "PuOr")
```

```{r}
pheatmap(noz_daytemp_degs_counts_match.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

Pull out just the sample count data:
```{r}
daytemp_degs_counts_match.heatmap2 <- select(daytemp_degs_counts_match.heatmap, CrabA_0, CrabB_0, CrabC_0,  CrabG_0, CrabH_0, CrabI_0, CrabA_2, CrabB_2, CrabC_2, CrabG_2, CrabH_2, CrabI_2)
head(daytemp_degs_counts_match.heatmap2)
```

```{r}
pheatmap(daytemp_degs_counts_match.heatmap2, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

## Do contrast between the different temperatures:
```{r}
deseq2.resTemp <- results(abcghi.dds,
                          contrast = c("temperature", "ambient",  "elevated"))
head(deseq2.resTemp)
```

```{r}
tmp <- deseq2.resTemp
# The main plot
plot(tmp$baseMean, tmp$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
tmp.sig <- deseq2.resTemp[!is.na(deseq2.resTemp$padj) & deseq2.resTemp$padj <= 0.05, ]
points(tmp.sig$baseMean, tmp.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(deseq2.resTemp[!is.na(deseq2.resTemp$padj) & deseq2.resTemp$padj <= 0.05, ])
```
2 DEGs 

```{r}
data <- plotCounts(abcghi.dds, which.min(deseq2.resTemp$pvalue),
                   intgroup = c("day","temperature"), returnData = TRUE)
ggplot(data, aes(x=day, y=count, color = temperature, group=temperature)) + 
  geom_point() + stat_smooth(se=FALSE,method="loess") + scale_y_log10()
```

## Try time series of just crab ABC genes over time: 

Subset those columns:
```{r}
abc <- select(indiv_counts, CrabA_0, CrabB_0, CrabC_0, CrabA_2, CrabB_2, CrabC_2, CrabA_17, CrabB_17, CrabC_17)
head(abc)
```

```{r}
abc.colData <- data.frame(crab=factor(c("crabA", "crabB", "crabC", "crabA", "crabB", "crabC", "crabA", "crabB", "crabC")), 
                             day=factor(c(rep("day0", 3), rep("day02", 3), rep("day17", 3))))
rownames(abc.colData) <- colnames(abc)
abc.dds <- DESeqDataSetFromMatrix(countData = abc,
                                     colData = abc.colData, 
                                     design = ~ day)
```

```{r}
abc.dds <- DESeq(abc.dds)
abc.res <- results(abc.dds)
abc.res$symbol <- mcols(abc.dds)$symbol
head(abc.res[order(abc.res$pvalue),],4)
```

```{r}
day <- abc.res
# The main plot
plot(day$baseMean, day$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
day.sig <- abc.res[!is.na(abc.res$padj) & abc.res$padj <= 0.05, ]
points(day.sig$baseMean, day.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(abc.res[!is.na(abc.res$padj) & abc.res$padj <= 0.05, ])
```
184

```{r}
data <- plotCounts(abc.dds, "TRINITY_DN1454_c0_g1_i8", 
                   intgroup=c("day","crab"), returnData=TRUE)
ggplot(data, aes(x=day, y=count, color=crab, group=crab)) + 
  geom_point() + stat_smooth(se=FALSE,method="loess") +  scale_y_log10()
```
^ Can plot only one gene at a time for this plot. Plot gene of interest. 


```{r}
resultsNames(abc.dds)
```

```{r}
betas <- coef(abc.dds)
colnames(betas)
```

```{r}
mat <- betas[, -1]
topGenes <- head(order(abc.res$pvalue),40)
colors <- RColorBrewer::brewer.pal(11, "PuOr")
pheatmap(mat [topGenes, ],  clustering_distance_rows = "euclidean", clustering_method = "average", treeheight_col = 80, legend = TRUE, color = colors, fontsize_col = 10, fontsize_row = 6)
```

## Try to figure out what genes are doing things over time in crabs ABC
Remove rows that have conts of '0' all across:
```{r}
abc_noz <- abc[rowSums(abc[, -1] >0) !=0, ]
head(abc_noz)
```
30,186 rows

Then, remove rows that have '0's in any sample: 
```{r}
row_sub = apply(abc_noz, 1, function(row) all(row !=0 ))
abc_nozs <- abc_noz[row_sub,]
head(abc_nozs)
```
502 rows

```{r}
abc_nozs.heatmap <- select(abc_nozs, "CrabA_0", "CrabA_2", "CrabA_17", "CrabB_0", "CrabB_2", "CrabB_17", "CrabC_0", "CrabC_2", "CrabC_17")
head(abc_nozs.heatmap)
```


```{r}
heatmapPurOr <- RColorBrewer::brewer.pal(11, "PuOr")
```

Create a heatmap of those count data: 
```{r}
pheatmap(abc_nozs.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurOr, fontsize_col = 20, fontsize_row = 12)
```



## Try all again, but with different individual samples: 

I'll go with A vs E time points 0 and 2 and 17. 
Subset those columns:
```{r}
#ave <- select(indiv_counts, CrabA_0, CrabA_2, CrabA_17, CrabE_0, CrabE_2, CrabE_17)
#head(ave)
```


```{r}
#ave.colData <- data.frame(temperature=factor(c(rep("ambient", 3), rep("decreased", 3))), 
#                             day=factor(c("day0", "day02", "day17", "day0", "day02", "day17")))
#rownames(ave.colData) <- colnames(ave)
#ave.dds <- DESeqDataSetFromMatrix(countData = ave,
#                                     colData = ave.colData, 
#                                     design = ~ temperature + day)
```

```{r}
#design(ave.dds) <- formula( ~ temperature + day)
#ave.dds <- DESeq(ave.dds)
#ave.res <- results(ave.dds)
#ave.res$symbol <- mcols(ave.dds)$symbol
#head(ave.res[order(ave.res$pvalue),],4)
```

```{r}
#ave.res <- ave.res
# The main plot
#plot(ave.res$baseMean, ave.res$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
#     xlab="mean of normalized counts",
#     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
#ave.res.sig <- ave.res[!is.na(ave.res$padj) & ave.res$padj <= 0.05, ]
#points(ave.res.sig$baseMean, ave.res.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
#abline(h=c(-1,1), col="blue")
```

```{r}
#write.table(ave.res.sig, "../analyses/DEGlist-ave-timeseries.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```


```{r}
#data <- plotCounts(ave.dds, which.min(ave.res$pvalue),
#                   intgroup = c("day","temperature"), returnData = TRUE)
#ggplot(data, aes(x=day, y=count, color = temperature, group=temperature)) + 
#  geom_point() + stat_smooth(se=FALSE,method="loess") + scale_y_log10()
```

```{r}
#ave.betas <- coef(ave.dds)
#colnames(ave.betas)
```


Don't understand this next chunk of code... why are only two columns showing? 
```{r}
#mat <- ave.betas[,-c(1,2)]
#mat[mat < -5] <- -5
#mat[mat > 5] <- 5
#topGenes <- head(order(ave.res$pvalue),10)
#colors <- RColorBrewer::brewer.pal(11, "PuOr")
#pheatmap(mat [topGenes, ],  clustering_distance_rows = "euclidean", clustering_method = "average", treeheight_col = 80, legend = TRUE, color = colors, fontsize_col = 10, fontsize_row = 6, ColV=FALSE)
```


