---
title: "DESeq_libraries8-11"
output: html_document
---
Rmd to go through `DESeq2` pipeline. Will check data. Then will perform some DEG analyses. 

Libraries 8-11 info: 
(These were used in the 02-DeSeq-Temperature.Rmd comparing elevated and decreased)     
They are all pooled samples:   
8 --> Day 2, infected, decreased temperature
9 --> Day 2, uninfected, decreased temperature
10 --> Day 2, infected, elevated temperature
11 --> Day 2, uninfected, elevated temperature

```{r}
library(DESeq2)
library(tidyverse)
```


Read in un-normalized counts:
FROM TRANSCRIPTOME V. 1.5
```{r}
pooldf <- read.table("../data/kallisto-0517.isoform.counts.txt", header = T, sep = "\t")
rownames(pooldf) <- pooldf$FeatureID
pooldf <- pooldf[,-1]
head(pooldf)
```

Rename columns so I know which samples are infected and which are uninfected:
```{r}
colnames(pooldf) <- c("uninfected_cold", "infected_cold", "uninfected_warm", "infected_warm")
head(pooldf)
```

## Get DEGs based on infection (not including temperature as part of the design)
Modeled after what was done in DESeq-infection.Rmd:
```{r}
colData <- data.frame(condition=factor(c("uninfected", "infected", "uninfected", "infected")), 
                             type=factor(rep("paired-end", 4)))
rownames(colData) <- colnames(pooldf)
dds <- DESeqDataSetFromMatrix(countData = pooldf,
                                     colData = colData, 
                                     design = ~ condition)
```

```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```


```{r}
head(res)
```
Sometimes a subset of the p values in res will be NA (“not available”). This is DESeq's way of reporting that all counts for this gene were zero, and hence not test was applied. In addition, p values can be assigned NA if the gene was excluded from analysis because it contained an extreme count outlier. For more information, see the outlier detection section of the vignette.       
(from https://www.bioconductor.org/help/course-materials/2015/LearnBioconductorFeb2015/B02.1.1_RNASeqLab.html#time)


```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```

So, without taking temperature into account, there are 428 DEGs between the infected and uninfected pooled samples. 

```{r}
infection_fig <- res
# The main plot
plot(infection_fig$baseMean, infection_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
infection_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(infection_fig.sig$baseMean, infection_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```



## Get DEGs based on infection with temperature as part of the design: 
From Multi-factor designs section of: 
http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#multi-factor-designs 

```{r}
deseq2.colData <- data.frame(condition=factor(c("uninfected", "infected", "uninfected", "infected")), 
                             type=factor(rep("paired-end", 4)),
                             temperature=factor(c("decreased", "decreased", "elevated", "elevated")))
rownames(deseq2.colData) <- colnames(pooldf)
deseq2.dds <- DESeqDataSetFromMatrix(countData = pooldf,
                                     colData = deseq2.colData, 
                                     design = ~ condition + temperature)
```

Check levels of temperature and condition (infection status)
```{r}
levels(deseq2.dds$temperature)
```

```{r}
levels(deseq2.dds$condition)
```

`DESeq2` automatically puts the levels in alphabetical order and the first listed level is the reference level for the factor. 

We want uninfected to be the reference. 
```{r}
deseq2.dds$condition = relevel(deseq2.dds$condition, "uninfected")
levels(deseq2.dds$condition)
```

Cool. 

The following will pull the results from `condition` (infection status) because that is our variable of interest. This tells us how temperature contributes to the infection DEGs
```{r}
design(deseq2.dds) <- formula(~ temperature + condition)
deseq2.dds <- DESeq(deseq2.dds)
```

Access results:
```{r}
deseq2.res <- results(deseq2.dds)
head(deseq2.res)
```

```{r}
summary(deseq2.res)
```


```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ])
```

```{r}
inf <- deseq2.res
# The main plot
plot(inf$baseMean, inf$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
inf.sig <- deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ]
points(inf.sig$baseMean, inf.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

## Now I need to figure out what these results mean: how exactly does temperature impact DEGs based on infection?


In the multifactor section of the `DESeq2` manual: 
The contrast argument of the function _results_ needs a character vector of three componenets: the name of the variable (in this case "temperature"), and the name of the factor level for the numerator of the log2 ratio (elevated) and the denomonator (decreased) 

A contrast is a linear combination of estimated log2 fold changes. Can be used to test if differences between groups are equal to zero. 
```{r}
resultsNames(deseq2.dds)
```


```{r}
deseq2.resTemp <- results(deseq2.dds,
                          contrast = c("temperature", "elevated",  "decreased"))
head(deseq2.resTemp)
```

```{r}
tmp <- deseq2.resTemp
# The main plot
plot(tmp$baseMean, tmp$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
tmp.sig <- deseq2.resTemp[!is.na(deseq2.resTemp$padj) & deseq2.resTemp$padj <= 0.05, ]
points(tmp.sig$baseMean, tmp.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(deseq2.resTemp[!is.na(deseq2.resTemp$padj) & deseq2.resTemp$padj <= 0.05, ])
```

```{r}
summary(deseq2.resTemp)
```

## Try similar thing to section above but with `name` argument of _results_
```{r}
deseq2.resTemp2 <- results(deseq2.dds,
                          name = "temperature_elevated_vs_decreased")
head(deseq2.resTemp2)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(deseq2.resTemp2[!is.na(deseq2.resTemp2$padj) & deseq2.resTemp2$padj <= 0.05, ])
```

Exact same results as what I did in previous section. 


```{r}
?results
```

## Notes:
From https://www.bioconductor.org/help/course-materials/2015/LearnBioconductorFeb2015/B02.1.1_RNASeqLab.html#de in section "multiple testing":     
- apparently I should be using a padj of 0.1 cut-off instead of 0.05 





### Attempt at running interaction term from `DESeq2` manual
Won't work. 
```{r}
#deseq2.dds$group <- factor(paste0(deseq2.dds$temperature, deseq2.dds$condition))
#design(deseq2.dds) <- ~ group
#deseq2.dds <- DESeq(deseq2.dds)
#resultsNames(deseq2.dds)
#results(deseq2.dds, contrast = c("group", "infecteddecrased", "infectedelevated"))
```
Error in checkForExperimentalReplicates(object, modelMatrix) : The design matrix has the same number of samples and coefficients to fit, so estimation of dispersion is not possible. Treating samples as replicates was deprecated in v1.20 and no longer supported since v1.22.













