---
title: "DESeq_libraries8-11"
output: html_document
---
Rmd to go through `DESeq2` pipeline. Will check data. Then will perform some DEG analyses. 

Libraries 8-11 info: 
(These were used in the 02-DeSeq-Temperature.Rmd comparing elevated and decreased)     
They are all pooled samples:   
8 --> Day 2, infected, decreased temperature
9 --> Day 2, uninfected, decreased temperature
10 --> Day 2, infected, elevated temperature
11 --> Day 2, uninfected, elevated temperature

```{r}
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
```


Read in un-normalized counts:
FROM TRANSCRIPTOME V. 1.5
```{r}
pooldf <- read.table("../data/kallisto-0517.isoform.counts.txt", header = T, sep = "\t")
rownames(pooldf) <- pooldf$FeatureID
pooldf <- pooldf[,-1]
head(pooldf)
```

Rename columns so I know which samples are infected and which are uninfected:
```{r}
colnames(pooldf) <- c("uninfected_cold", "infected_cold", "uninfected_warm", "infected_warm")
head(pooldf)
```

## Get DEGs based on infection (not including temperature as part of the design)
Modeled after what was done in DESeq-infection.Rmd:
```{r}
#colData <- data.frame(condition=factor(c("uninfected", "infected", "uninfected", "infected")), 
#                             type=factor(rep("paired-end", 4)))
#rownames(colData) <- colnames(pooldf)
#dds <- DESeqDataSetFromMatrix(countData = pooldf,
#                                     colData = colData, 
#                                     design = ~ condition)
```

```{r}
#dds <- DESeq(dds)
#res <- results(dds)
#res <- res[order(rownames(res)), ]
```


```{r}
#head(res)
```
Sometimes a subset of the p values in res will be NA (“not available”). This is DESeq's way of reporting that all counts for this gene were zero, and hence not test was applied. In addition, p values can be assigned NA if the gene was excluded from analysis because it contained an extreme count outlier. For more information, see the outlier detection section of the vignette.       
(from https://www.bioconductor.org/help/course-materials/2015/LearnBioconductorFeb2015/B02.1.1_RNASeqLab.html#time)


```{r}
# Count number of hits with adjusted p-value less then 0.05
#dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
[1] 428   6

So, without taking temperature into account, there are 428 DEGs between the infected and uninfected pooled samples. 

```{r}
#infection_fig <- res
# The main plot
#plot(infection_fig$baseMean, infection_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
#     xlab="mean of normalized counts",
#     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
#infection_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
#points(infection_fig.sig$baseMean, infection_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
#abline(h=c(-1,1), col="blue")
```



## Get DEGs based on infection with temperature as part of the design: 
Taking temperature exposure into account!!      
From Multi-factor designs section of:       
http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#multi-factor-designs 

```{r}
deseq2.colData <- data.frame(condition=factor(c("uninfected", "infected", "uninfected", "infected")), 
                             type=factor(rep("paired-end", 4)),
                             temperature=factor(c("decreased", "decreased", "elevated", "elevated")))
rownames(deseq2.colData) <- colnames(pooldf)
deseq2.dds <- DESeqDataSetFromMatrix(countData = pooldf,
                                     colData = deseq2.colData, 
                                     design = ~ condition + temperature)
```

Check levels of temperature and condition (infection status)
```{r}
levels(deseq2.dds$temperature)
```

```{r}
levels(deseq2.dds$condition)
```

`DESeq2` automatically puts the levels in alphabetical order and the first listed level is the reference level for the factor. 

We want uninfected to be the reference. 
```{r}
deseq2.dds$condition = relevel(deseq2.dds$condition, "uninfected")
levels(deseq2.dds$condition)
```

Cool. 

The following will pull the results from `condition` (infection status) because that is our variable of interest. This tells us how temperature contributes to the infection DEGs
```{r}
design(deseq2.dds) <- formula(~ temperature + condition)
deseq2.dds <- DESeq(deseq2.dds)
```

Access results:
```{r}
deseq2.res <- results(deseq2.dds)
head(deseq2.res)
```

```{r}
summary(deseq2.res)
```


```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ])
```

```{r}
inf <- deseq2.res
# The main plot
plot(inf$baseMean, inf$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
inf.sig <- deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ]
points(inf.sig$baseMean, inf.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

```{r}
#write.table(inf.sig, "../analyses/DEGlist-infection_temp-libs8-11.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
Already wrote out, so I'm commenting above script out. 

## Now I need to figure out what these results mean: how exactly does temperature impact DEGs based on infection?


In the multifactor section of the `DESeq2` manual: 
The contrast argument of the function _results_ needs a character vector of three componenets: the name of the variable (in this case "temperature"), and the name of the factor level for the numerator of the log2 ratio (elevated) and the denominator (decreased) 

A **contrast** is a linear combination of estimated log2 fold changes. Can be used to test if differences between groups are equal to zero. 
```{r}
resultsNames(deseq2.dds)
```


```{r}
deseq2.resTemp <- results(deseq2.dds,
                          contrast = c("temperature", "elevated",  "decreased"))
head(deseq2.resTemp)
```

```{r}
tmp <- deseq2.resTemp
# The main plot
plot(tmp$baseMean, tmp$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
tmp.sig <- deseq2.resTemp[!is.na(deseq2.resTemp$padj) & deseq2.resTemp$padj <= 0.05, ]
points(tmp.sig$baseMean, tmp.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(deseq2.resTemp[!is.na(deseq2.resTemp$padj) & deseq2.resTemp$padj <= 0.05, ])
```

```{r}
summary(deseq2.resTemp)
```

```{r}
#write.table(tmp.sig, "../analyses/DEGlist-contrast_temp-libs8-11.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
Already wrote out, so I'm commenting above script out. 

## Notes:
From https://www.bioconductor.org/help/course-materials/2015/LearnBioconductorFeb2015/B02.1.1_RNASeqLab.html#de in section "multiple testing":     
- apparently I should be using a padj of 0.1 cut-off instead of 0.05 ..???


**DEGs (183) based on infection taking temperature into account amd DEGs (130) just between elevated and decreased temperature treatments are the results.** 

## After chat with SR --> `join` count data with list of 183 AND 130 genes to see what makes sense. 

### `join` count data with list of 183

Read in DEGlist of 183 (.txt version where the rownames of Trinity IDs are a column called "Trinity_ID": 
```{r}
deglist_inftemp <- read.delim("../analyses/DEGlist-infection_temp-libs8-11.txt", sep = '\t')
```
183 rows --> good to go. 

Make row names of pooldf Trinity IDs into a column called "Trinity_ID":
```{r}
#make Trinity IDs into a column in tmdf
df <- pooldf
names <- rownames(pooldf)
rownames(df) <- NULL
data <- cbind(names, df)

#rename column 1 in tmdf
colnames(data) <- c("Trinity_ID", "uninfected_cold", "infected_cold", "uninfected_warm", "infected_warm")
head(data)
```

`join` based on Trinity ID
```{r}
degs_inftemp_counts <- left_join(data, deglist_inftemp, by = "Trinity_ID")
head(degs_inftemp_counts)
```

Only want the ones that match!
```{r}
inftempdegs_counts_match <- filter(degs_inftemp_counts, baseMean != "NA")
head(inftempdegs_counts_match)
```
183 rows. 

Write out list with count data:
```{r}
#write.table(inftempdegs_counts_match, "../analyses/DEGlist-inf-temp-libs8-11_counts.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```
already wrote out - comment out script

### `join` count data with list of 130

Read in DEGlist of 130 (.txt version where the rownames of Trinity IDs are a column called "Trinity_ID": 
```{r}
degs_contasttemp <-  read.delim("../analyses/DEGlist-contrast_temp-libs8-11.txt", sep = '\t')
```
130 rows - excellent

`join` based on Trinity ID
```{r}
deg_contrasttemp_counts <- left_join(data, degs_contasttemp, by = "Trinity_ID")
head(deg_contrasttemp_counts)
```

Only want the ones that match!
```{r}
contrtempdegs_counts_match <- filter(deg_contrasttemp_counts, baseMean != "NA")
head(contrtempdegs_counts_match)
```
130 rows 

Values in that list with large positive log2 fold changes only have counts in warm samples

values in that list with small negative log2 fold changes only have counts in cold samples

Write out list with count data:
```{r}
#write.table(contrtempdegs_counts_match, "../analyses/DEGlist-contrast-temp-libs8-11_counts.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```
already wrote out - comment out script


## Make heatmaps of the two lists of count datas

### Heatmap of counts from 183 list (infection comparison taking temperature into account)

Set Trinity_ID as row names:
```{r}
rownames(inftempdegs_counts_match) <- inftempdegs_counts_match$Trinity_ID #set Trinity IDs as rownames
inftempdegs_counts_match.heatmap <- inftempdegs_counts_match[,-1] #remove redundant column
head(inftempdegs_counts_match.heatmap)
```

Pull out just the sample count data:
```{r}
inftempdegs_counts_match.heatmap <- select(inftempdegs_counts_match.heatmap, uninfected_cold, infected_cold, uninfected_warm, infected_warm)
head(inftempdegs_counts_match.heatmap)
```

Going to remove all rows that have "0" for all samples: 
```{r}
noz_inftempdegs_counts_match.heatmap <- inftempdegs_counts_match.heatmap[rowSums(inftempdegs_counts_match.heatmap[, -1] >0) !=0, ]
head(noz_inftempdegs_counts_match.heatmap)
```

```{r}
heatmapPurpleOrange <- RColorBrewer::brewer.pal(11, "PuOr")
```

```{r}
pheatmap(noz_inftempdegs_counts_match.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

### Heatmap of counts from 130 list (contrast of temperatures from infection comparison taking temperature into account)

Set Trinity_ID as row names:
```{r}
rownames(contrtempdegs_counts_match) <- contrtempdegs_counts_match$Trinity_ID #set Trinity IDs as rownames
contrtempdegs_counts_match.heatmap <- contrtempdegs_counts_match[,-1] #remove redundant column
head(contrtempdegs_counts_match.heatmap)
```

Pull out just the sample count data:
```{r}
contrtempdegs_counts_match.heatmap <- select(contrtempdegs_counts_match.heatmap, uninfected_cold, infected_cold, uninfected_warm, infected_warm)
head(contrtempdegs_counts_match.heatmap)
```

Going to remove all rows that have "0" for all samples: 
```{r}
noz_contrtempdegs_counts_match.heatmap <- contrtempdegs_counts_match.heatmap[rowSums(contrtempdegs_counts_match.heatmap[, -1] >0) !=0, ]
head(noz_contrtempdegs_counts_match.heatmap)
```

```{r}
pheatmap(noz_contrtempdegs_counts_match.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

## Get the genes that correspond with the clusters (the purple ones --> higher expression)

### Gene clusters from heatmap comparing infection status taking temperature into account (first heatmap)

Set the heatmap as an object called 'results':
```{r}
results <- pheatmap(noz_inftempdegs_counts_match.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

extract 4 clusters. 
```{r}
inftemp.clust <- cbind(noz_inftempdegs_counts_match.heatmap,
                   cluster = cutree(results$tree_row,
                                     k = 4))
head(inftemp.clust)
```

### Gene clusters from heatmap contrasting temperatures from the comparison infection status taking temperature into account (second heatmap)

Set the heatmap as an object called 'results2':
```{r}
results2 <- pheatmap(noz_contrtempdegs_counts_match.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

extract 4 clusters. 
```{r}
contrasttemp.clust <- cbind(noz_contrtempdegs_counts_match.heatmap,
                   cluster = cutree(results2$tree_row,
                                     k = 4))
head(contrasttemp.clust)
```

## `join` count data files with cluster numbers with the `DESeq2` results, and `blast` annotation and GO annotation (from transcirptome v. 1.5)

### `join` the transcriptome v. 1.5 `blast` output with GO annotations: 

upload `blast` output from transcriptome v. 1.5
```{r}
blastout <- read.delim("../../project-crab/analyses/BLAST_to_GOslim/_blast-sep.tab", sep = '\t', header = FALSE)
head(blastout)
```
Rename some columns so that this file can be `join`-ed with the uniprot-SP-Go:
```{r}
colnames(blastout) <- c("Trinity_ID", "V2", "uniprot_acc_ID", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
head(blastout)
```
Read in the uniprot-SP-GO.sorted file:
```{r}
uniprot_sp_GO <- read.delim("http://owl.fish.washington.edu/halfshell/bu-alanine-wd/17-07-20/uniprot-SP-GO.sorted", sep = '\t', header = FALSE)
head(uniprot_sp_GO)
```

Rename some columns: 
```{r}
colnames(uniprot_sp_GO) <- c("uniprot_acc_ID", "entry_name", "review_status", "protein_name", "gene_name", "organism", "length", "V8", "V9", "V10", "V11", "gene_ontology_IDs")
head(uniprot_sp_GO)
```

`join` blastout and uniprot_sp_GO by "uniprot_acc_ID" column:
```{r}
blastout_uniprot <- left_join(blastout, uniprot_sp_GO, by = "uniprot_acc_ID")
head(blastout_uniprot)
```




### List of 183
Set Trinity IDs as a column called "Trinity_ID":
```{r}
inftemp.res.clusts4 <- inftemp.clust
Trinity_ID <- rownames(inftemp.res.clusts4)
rownames(inftemp.res.clusts4) <- NULL
final_inftemp.res.clusts4 <- cbind(Trinity_ID, inftemp.res.clusts4)
head(final_inftemp.res.clusts4)
```

Subset the columns Trinity_ID and cluster:
```{r}
inftemp.trinID.clust <- select(final_inftemp.res.clusts4, "Trinity_ID", "cluster")
head(inftemp.trinID.clust)
```

`join` with contrtempdegs_counts_match
```{r}
inftemp.clust.degs <- left_join(inftempdegs_counts_match, inftemp.trinID.clust, by = "Trinity_ID")
head(inftemp.clust.degs)
```

`join` with uniprot-GO-blastout: 
```{r}
annot_inftemp.clust.degs <- left_join(inftemp.clust.degs, blastout_uniprot, by = "Trinity_ID")
head(annot_inftemp.clust.degs)
```

### List of 130
Set Trinity IDs as a column called "Trinity_ID":
```{r}
contrtemp.res.clusts4 <- contrasttemp.clust
Trinity_ID <- rownames(contrtemp.res.clusts4)
rownames(contrtemp.res.clusts4) <- NULL
final_contrtemp.res.clusts4 <- cbind(Trinity_ID, contrtemp.res.clusts4)
head(final_contrtemp.res.clusts4)
```

Subset the Trinity_ID column with the cluster column:
```{r}
contrtemp.trinID.clust <- select(final_contrtemp.res.clusts4, "Trinity_ID", "cluster")
head(contrtemp.trinID.clust)
```


`join` with contrtempdegs_counts_match
```{r}
contrtemp.clust.degs <- left_join(contrtempdegs_counts_match, contrtemp.trinID.clust, by = "Trinity_ID")
head(contrtemp.clust.degs)
```

`join` with uniprot-GO-blastout: 
```{r}
annot_contrtemp.clust.degs <- left_join(contrtemp.clust.degs, blastout_uniprot, by = "Trinity_ID")
head(annot_contrtemp.clust.degs)
```






