---
title: "WGCNA-indiv-crabs"
output: html_document
---
Rmd to perform WGCNA with individual crab RNAseq datasets (transcriptome v 3.1). 

Based on tutorials from WGCNA package and script by Yaamini V. : https://github.com/eimd-2019/project-EWD-transcriptomics/blob/master/analyses/WGCNA/WGCNA.md

https://www.biostars.org/p/280650/     
According to the above on bioconductor, I will:    
- remove counts that are consistently low (remove those that have a count less than 10)          
- Transform data --> either vst (`DESeq2` package), or use normalized counts and log-transform using log2(x+1)           
 
Install packages:
```{r}
#install.packages("dichromat")
```
 
Load packages:         
```{r}
library(dplyr)
library(tidyverse)
library(DESeq2)
library(WGCNA)
library(data.table)
library(dichromat)
```

```{r}
options(stringsAsFactors = FALSE) #Important setting to include from tutorial
```

## Obtain session info:
```{r}
sessionInfo()
```

## Format Data
### Read in logCPM count data from individual crab RNAseq (transcriptome v 3.1):      
File was created in Rmd called "EdgeR-indiv.crabs-RNAseq.Rmd"
```{r}
crablogCPM <- read.delim("../analyses/ind.crabs.logCPM.txt", header = TRUE)
head(crablogCPM)
```

Set rownames as a column:     
```{r}
crablogCPM <- tibble::rownames_to_column(crablogCPM, "Trinity_ID")
head(crablogCPM)
```


Transpose dataframe so that samples are rows and genes are columns:
```{r}
crabExpr0 <- as.data.frame(t(crablogCPM[,-1]))
names(crabExpr0) <- crablogCPM$Trinity_ID
head(crabExpr0)
```

### Check for missing values and outliers:
```{r}
gsg <- goodSamplesGenes(crabExpr0, verbose = 3) #check for genes and samples with too many missing values
```
```{r}
gsg$allOK #TRUE = all genes have passed and can continue
```

```{r}
sampleTree <- hclust(dist(crabExpr0), method = "average") #create dendrogram to visually inspect for outliers
plot(sampleTree)
```
Crab D 17 and Crab B 2 are obvious outliers.          
Crab D 0 and Crab A 17 are also kind of outliers... 

I'll just keep going and come back to this. will likely remove outliers. 

### Create trait data matrix:    
Eelgrass project just has 2 differences between samples: exposed to Laby and unexposed. 

This project has time (in sample name), infection status... all except crab E are infected, and temperature - 3 at ambient, 3 at decreased, and 3 at elevated... How should I do this data matrix? 

I made a .csv in excel called "traits". read it in: 
```{r}
crabtraits <- read.csv("../analyses/crab.indiv.traits.csv")
head(crabtraits)
```
Make column "X" the rownames: 
Set Trinity_ID as row names:
```{r}
rownames(crabtraits) <- crabtraits$X #set Trinity IDs as rownames
crabtraits <- crabtraits[,-1] #remove redundant column
head(crabtraits)
```

```{r}
sampleTree2 <- hclust(dist(crabExpr0), method = "average") #create dendrogram to look at sample and trait clustering
traitColors <- numbers2colors(crabtraits, signed = FALSE) #convert traits to color values
plotDendroAndColors(sampleTree2, traitColors,
                    groupLabels = names(crabtraits)) #plot dendrogram
```

Save expression and trait data:
```{r}
save(crabExpr0, crabtraits, file = "../analyses/indiv.crab-WGCNA-dataInput.RData")
```

### Network connection and module detection

#### Determine soft-thresholding power
load data input from previous section:
```{r}
#cnames <- load(file = "../analyses/indiv.crab-WGCNA-dataInput.RData")
```

```{r}
#powers <- c(c(1:10), seq(from = 12, to = 20, by = 2)) #create a set of soft-thresholding powers
#sft <- pickSoftThreshold(crabExpr0, powerVector = powers, verbose = 5) #use the network topology analysis funciton to evaluate soft-thresholding power values
```
COMPUTER crashed with the above code, so I'll comment it out. 



### Topological Overlap Matrix (TOM)












